### ISSUE TITLE
Bump Google.Protobuf dependency on System.Memory to version >=4.5.3

### ISSUE THREAD

**Issue by kskalski** [protocolbuffers/protobuf#7848]
**What language does this apply to?**
C#/.NET

**Describe the problem you are trying to solve.**
With version 3.13.0 Google.Protobuf raised dependency on netstandard from >=1.0 to >=1.1 and added dependency on System.Memory 4.0.1.0 (provided by package 4.5.2) - previously this dependency was only present in netstandard2.0 build.

The dependency itself isn't a problem, it's a pretty common package, however the currently used version is a bit old and conflicts with other packages (e.g. Grpc.Core) causing problems upstream, e.g. 
https://github.com/grpc/grpc/issues/22251 or https://www.gitmemory.com/issue/grpc/grpc/21908/585508325

In my toolset it causes error like this during compilation:
```
error : Can't load types from 'Google.Protobuf, Version=3.13.0.0, Culture=neutral, PublicKeyToken=a7d26565bac4d604'. Error: Nie można załadować pliku lub zestawu '
System.Memory, Version=4.0.1.0, Culture=neutral, PublicKeyToken=cc7b13ffcd2ddd51' lub jednej z jego zależności. Nie można odnaleźć określonego pliku..
```

**Describe the solution you'd like**
I think matching System.Memory dependency with Grpc by bumping it to 4.0.1.1 (provided by package 4.5.3 or newest 4.5.4) would be desirable. I'm not completely sure about compatibility implications of such change (for netstandard1.1), so a safer plan would be to do this only for netstandard2.0 build, for which version 4.5.2 (currently used) and 4.5.3 of System.Memory do not differ on dependencies:
```
System.Buffers (>= 4.4.0)
System.Numerics.Vectors (>= 4.4.0)
System.Runtime.CompilerServices.Unsafe (>= 4.5.2)
```

**Describe alternatives you've considered**
Right now I'm using Google.Protobuf 3.12.4 netstandard1.0 build + Grpc.Core 2.31.0 and they work find together. However trying to upgrade Google.Protobuf to 3.13.0 creates a conflict, downgrading System.Memory to 4.5.2 causes problem with Grpc.Core 

**Additional context**
My .NET compiler for some reason doesn't allow using newer System.Memory assembly with Google.Protobuf 3.13.0, though I think some people got past compilation phase and got a warning at runtime instead, which can be further improved by using binding redirect.

**Comment by jtattermusch** [comment]
Adding a `<AutoGenerateBindingRedirects>true</AutoGenerateBindingRedirects>` to your .csproj should help.
http://go.microsoft.com/fwlink/?LinkId=294190

What TFM is your project targeting? Are you using classic .csproj project or the new .NET core .csproj format?

**Comment by kskalski** [comment]
The particular project / compiler that I'm using doesn't support ``AutoGenerateBindingRedirects`` thus I rely on compatible dependencies. I'm targeting net45 here, but whether I pick ``netstandard1.1``, ``netstandard2.0`` or ``net45`` ``Google.Protobuf.dll`` version as dependency I still get the error, since all of them depend on the same 4.0.1.0 ``System.Memory`` that is different than grpc's dependency.

**Comment by SteveCruise** [comment]
Please upgrade System.Memory to 4.5.3 in Google.Protobuf.dll

**Comment by SteveCruise** [comment]
I solved it.
Download google protobuf source, enter csharp dir, change `<PackageReference Include="System.Memory" Version="4.5.3" />` from 4.5.2 to 4.5.3, run `dotnet publish -c Release -f net45`, replace Google.Protobuf.dll. Thank you for every contributor, you make the world better!

**Comment by jtattermusch** [comment]
This will be fixed once #7940 is merged. Alternatively, it can be solved without needing to upgrade anything by adding this line to your .csproj:
https://github.com/grpc/grpc/blob/9fac907f806c3a9b9b775b15d8e3cb387c4d7048/test/distrib/csharp/DistribTest/DistribTest.csproj#L67
